guides-markdown/crowdfund-tx.md

```diff
@@ -0,0 +1,487 @@
+  // In reality you might get these coins from a wallet.
+  coins[i] = [Coin.fromTX(cb, 0, -1)];
+}
+```
+
+The coins object you've created above should look something like this:
+
+```json
+{
+  "0":
+      [
+        {
+          "type": "pubkeyhash",
+          "version": 1,
+          "height": -1,
+          "value": "5.0",
+          "script": <Script: OP_DUP OP_HASH160 0x14 0x64cc4e55b2daec25431bd879ef39302a77c1c1ce OP_EQUALVERIFY OP_CHECKSIG>,
+          "coinbase": true,
+          "hash": "151e5551cdcec5fff06818fb78ac3d584361276e862b5700110ec8321869d650",
+          "index": 0,
+          "address": <Address: type=pubkeyhash version=-1 str=mphvcfcFneRZvyYsmzhy57cSDzFbGrWaRb>
+        }
+      ],
+  "1": [...]
+}
+```
+
+#### Step 4: Prepare your Coins
+Above, we just funded our funder accounts with a single 5BTC outpoint. This means that the next transaction funded from these accounts can only use that one outpoint (or *coin*) as an input and send the remainder back as change. Remember, in Bitcoin the way you send funds is you fund a transaction with a full UTXO (in this case we only have one worth 5BTC available to our keychains) and then send the change back to yourself as an additional output. Since ALL|ANYONECANPAY transactions mean a fixed output, you can't add new change outputs without other signatures becoming invalid which means we need a coin available equal to the amount we want to contribute to the crowdfund.
+
+So what we want to do is have each funder create a coin (UTXO) with the value of what they want to donate.
+
+Let's create an asyncronous utility function to help us split the coinbases we created in the previous step (we use the `fund` method on the `MTX` primitive to do this, which is asynchronous).
+
+```javascript
+const splitCoinbase = async function splitCoinbase(keyrings, coins, targetAmount, txRate) {
+
+  // loop through each coinbase coin to split
+  for(const coinsIndex in coins) {
+    // funder will be at the same index as the key of the coins we are accessing
+    const funderKeyring = keyrings[coinsIndex];
+    const mtx = new MTX();
+
+    assert(coins[coinsIndex][0].value > targetAmount, 'coin value is not enough!');
+
+    // create a transaction that will have an output equal to what we want to fund
+    mtx.addOutput({
+      address: funderKeyring.getAddress(),
+      value: targetAmount
+    });
+
+    // shift off the coinbase coin to use to fund the splitting transaction
+    // the fund method will automatically split the remaining funds to the change address
+    await mtx.fund([coins[coinsIndex].shift()], {
+      rate: txRate,
+      // send change back to an address belonging to the funder
+      changeAddress: funderKeyring.getAddress()
+    }).then(() => {
+      // sign the mtx to finalize split
+      mtx.sign(funderKeyring);
+      assert(mtx.verify());
+
+      const tx = mtx.toTX();
+      assert(tx.verify(mtx.view));
```

Comment:
Why do you need to verify as TX too? (They call the same thing)

Reply:
I took this from the "working with transactions guide" that JJ created in the bcoin repo. Maybe it's checking just in case something went wrong when converting it to the TX? I'm guessing there's something else that gets changed in the tx when making it immutable that maybe needs to be checked after